app/controllers/api/v1/reviews_controller.rb

```diff
@@ -0,0 +1,21 @@
+class Api::V1::ReviewsController < ApiController
+  def index
+    product = Product.find(params[:product_id])
+    reviews = product.reviews
+    reviewwithuser = []
+    reviews.each do |review| 
+      newreview = {rating: review.rating,
+      body: review.body,
+      username: User.find(review.user_id)
+      }
+    reviewwithuser << newreview
+    end
+    render json: reviewwithuser
+  end
+
+  private
+  def review_params
+    params.require(:review).permit(:rating, :body)
+  end
+
```

Comment:
Couple things here: 
- add a line after line 1 
- add a line after line 16
- create a review serializer for your reviews index that also adds the current user, instead of doing all of the stuff I see on lines 6 - 11
- indentation is a bit odd on lines 6 - 11 anyways. 

I am willing to merge this without a serializer to move forward, but your entire live will be made much simpler with one, and you would still need to address the other changes.

Reply:
**Scratch everything I said above**

You've done good work, but the more optimal implementation is to add all of this review data to the fetch for the product show page. 

This will be a bit of a refactor, but a necessary one. So there should only be one `useEffect` fetch request in `ProductShowContainer`, specifically `fetch(`/api/v1/products/${productId}`)`, making this endpoint obsolete.